content/openapiv4.md

```diff
@@ -0,0 +1,212 @@
+    },
+    "emission_date": "",
+    "pay_date": "",
+    "status": "",
+    "currency": "",
+    "price_factor": float,
+    "discount": float,
+    "vat_percent": float,
+    "total_tax_excluded": {
+      "currency": "",
+      "amount": float,
+      "amount_formatted": "",
+      "default_display": ""
+    },
+    "total_tax": {
+      "currency": "",
+      "amount": float,
+      "amount_formatted": "",
+      "default_display": ""
+    },
+    "category": "",
+    "payment_provider": "",
+    "transaction_id": "",
+    "customer_order_id": ""
+  },
+  {...}
+]
+```
+
+## Deployments
+- `/v4/orchestration/organisations/{ownerId}/applications/{applicationId}/deployments`
+- `/v4/orchestration/organisations/{ownerId}/applications/{applicationId}/deployments/{deploymentId}`
+- Type: `GET`
+
+Example response, for each deployment:
+```bash
+[
+  {
+    "id": "",
+    "ownerId": "",
+    "applicationId": "",
+    "startDate": "",
+    "state": "",
+    "steps": [
+      {
+        "state": "",
+        "date": ""
+      },
+      {
+        "state": "",
+        "date": ""
+      },
+      {
+        "state": "",
+        "date": ""
+      }
+    ],
+    "version": {
+      "commitId": ""
+    },
+    "origin": {
+      "action": "",
+      "cause": "",
+      "source": "",
+      "authorId": "",
+      "constraints": [
+        ""
+      ],
+      "priority": ""
+    },
+    "hasDedicatedBuild": boolean
+  },
+  {...}
+]
+```
+
+## Instances
+- `/v4/orchestration/organisations/{ownerId}/applications/{applicationId}/instances`
+- `/v4/orchestration/organisations/{ownerId}/applications/{applicationId}/deployments/{deploymentId}/instances`
+- Type: `GET`
+
+Example response, for each instance:
+```bash
+[
+  {
+    "id": "",
+    "ownerId": "",
+    "applicationId": "",
+    "deploymentId": "",
+    "name": "",
+    "flavor": "",
+    "index": int,
+    "state": "",
+    "creationDate": "",
+    "deletionDate": "",
+    "network": {
+      "ip": "",
+      "port": int
+    },
+    "isBuildVm": boolean
+  },
+  {...}
+]
+```
+
+## Load Balancers
+- `/v4/load-balancers/organisations/{ownerId}/applications/{applicationId}/load-balancers/default`
+- Type: `GET`
+
+Example response:
+```bash
+[
+  {
+    "id": "",
+    "name": "",
+    "zoneId": "",
+    "dns": {
+      "cname": "",
+      "a": []
+    }
+  }
+]
+```
+
+## Logs
+- `/v4/logs/organisations/{ownerId}/applications/{applicationId}/logs`
+- Type: `GET`
+
+Answers with logs in a SSE (Server-Sent Events) stream. You can use the optional parameters to filter it:
+
+| Parameter      | Type     | Comment            |
+| :---           | :---     | :---               | 
+| `since`        | `string` | ISO 8601 formatted |
+| `until`        | `string` | ISO 8601 formatted |
+| `limit`        | `int64 ` | Minimum: `1`       |
+| `filter`       | `string` |                    |
+| `instanceId`   | `string` |                    |
+| `deploymentId` | `string` |                    |
```

Comment:
"since".as(stringToOffsetDateTime).optional,
"until".as(stringToOffsetDateTime).optional,
"limit".as[Int].optional,
"deploymentId".as(DeploymentID.stringUnmarshaller).optional,
"instanceId".as(InstanceID.stringUnmarshaller).repeated,
"filter".optional,
"service".as[String].repeated, (special all value and default to "bas", "bas-deploy")
"field".as[String].repeated,
"throttleElements".as[Int].withDefault(ovd.logs.sse.throttleElements),
"throttlePerInMilliseconds".as[Long].withDefault(ovd.logs.sse.throttlePer.toMillis)